src/index.ts

```diff
@@ -44,13 +77,20 @@ export * from './errors';
  * @throws {StrapiInitializationError} If the provided baseURL doesn't conform to a valid HTTP or HTTPS URL,
  *                                        or if the auth configuration is invalid.
  */
-export const strapi = (config: StrapiConfig) => {
-  const configValidator = new StrapiConfigValidator();
-
-  return new Strapi<typeof config>(
-    // Properties
-    config,
-    // Dependencies
-    configValidator
-  );
+export const strapi = (config: Config) => {
+  const { baseURL, auth } = config;
+
+  const sdkConfig: StrapiConfig = { baseURL };
+
+  // In this factory, to keep things simple, users can't manually set the strategy options.
+  // Since the SDK constructor needs to define a proper strategy,
+  // it is handled here if the auth property is provided
+  if (auth !== undefined) {
```

Comment:
Should we just check if this is a string?

Reply:
This was my initial idea, but the actual validation is done down the line in the `SDK > Auth > Provider` itself. So, to keep nice error messages, I was gonna let this be propagated

Reply:
By doing this we basically end up with a nice StrapiInitError with StrapiValidationError (+ "wrong type for token" as error message) instead of just throw an error